Dockerfile

```
@@ -1,39 +1,29 @@
 FROM python:3.8
-ENV PYTHONUNBUFFERED 1
+
+ENV PYTHONUNBUFFERED=1 PYTHONHASHSEED=random \
+    PYTHONDONTWRITEBYTECODE=1 PIP_NO_CACHE_DIR=1
 
 ARG REDIS='external'
 
-# Install supervisord
-RUN apt-get update && apt-get install -y supervisor
+# optionally install REDIS
 RUN if [ "${REDIS}" != "external" ]; then apt-get update && apt-get install -y redis-server; fi
 
-RUN mkdir -p /var/log/supervisor
-
-# Allows docker to cache installed dependencies between builds
-COPY requirements.txt /tmp/pip-tmp/requirements.txt
-# RUN pip3 --disable-pip-version-check --no-cache-dir install -r /tmp/pip-tmp/requirements.txt \
-#    && rm -rf /tmp/pip-tmp
-run pip install -r /tmp/pip-tmp/requirements.txt
-run rm -rf /tmp/pip-tmp
+# install supervisord (supervisor-stdout is not py3 compatible in pypi)
+RUN pip install supervisor git+https://github.com/coderanger/supervisor-stdout
 
-COPY supervisord.conf /tmp/supervisord-tmp/supervsiord.conf
-COPY supervisord-redis.conf /tmp/supervsiord-tmp/supervisord-redis.conf
-RUN if [ "${REDIS}" != "external" ]; then cp /tmp/supervsiord-tmp/supervisord-redis.conf /etc/supervisord.conf; else cp /tmp/supervisord-tmp/supervsiord.conf /etc/supervisord.conf; fi
+# Sets work directory to /code
+WORKDIR /code
 
+# Allows docker to cache installed dependencies between builds
+COPY requirements.txt .
+RUN pip install -r requirements.txt
 
 # Adds our application code to the image
 COPY . /code
 
-# Adds loadenv to profile.d
-COPY ./scripts/loadenv.sh /root/.bashrc
+# configure for redis
+RUN if [ "${REDIS}" != "external" ]; then cp supervisord-redis.conf /etc/supervisord.conf; else cp supervsiord.conf /etc/supervisord.conf; fi
```

```suggestion
RUN if [ "${REDIS}" != "external" ]; then cp supervisord-redis.conf /etc/supervisord.conf; else cp supervisord.conf /etc/supervisord.conf; fi
```

Fixed typo for the case of external redis